weblate/trans/migrations/0045_component_file_format_params.py

```diff
@@ -0,0 +1,178 @@
+        "weblate.gettext.customize",
+        "weblate.yaml.customize",
+    ]
+    Addon.objects.filter(name__in=to_delete).delete()
+
+    # remove configurations from msgmerge addon
+    Addon.objects.filter(name="weblate.gettext.msgmerge").update(configuration={})
+
+
+def convert_addon_config_to_file_format_params(addon) -> tuple[tuple, dict[str, Any]]:
+    """Convert addon configuration to file format parameters."""
+    attrs = ADDON_CONFG_TO_FILE_FORMAT_PARAMS[addon.name]
+    params = {}
+    for addon_config, param_name in attrs["format_params"].items():
+        if addon_config in addon.configuration:
+            params[param_name] = addon.configuration[addon_config]
+    return attrs["file_formats"], params
+
+
+def collect_addon_configurations(apps) -> dict[str, list | dict]:
+    """Collect and organize addon configurations by scope."""
+    Addon = apps.get_model("addons", "Addon")
+
+    addons = Addon.objects.filter(
+        name__in=ADDON_CONFG_TO_FILE_FORMAT_PARAMS.keys()
+    ).select_related("component", "project")
+
+    site_wide = []
+    project_wide = defaultdict(list)
+    component_wide = defaultdict(list)
+
+    for addon in addons:
+        file_formats, params = convert_addon_config_to_file_format_params(addon)
+        config_entry = (file_formats, params)
+
+        if not addon.component and not addon.project:
+            site_wide.append(config_entry)
+        elif addon.project and not addon.component:
+            project_wide[addon.project_id].append(config_entry)
+        elif addon.component:
+            component_wide[addon.component_id].append(config_entry)
+    return {
+        "site_wide": site_wide,
+        "project_wide": project_wide,
+        "component_wide": component_wide,
+    }
+
+
+def merge_file_format_params(component, configs) -> dict[str, bool | int | str]:
+    """Merge file format parameters for a component based on priority."""
+    file_format_params = {}
+    for file_formats, params in configs["site_wide"]:
+        if component.file_format in file_formats:
+            file_format_params.update(params)
+
+    for file_formats, params in configs["project_wide"].get(component.project_id, []):
+        if component.file_format in file_formats:
+            file_format_params.update(params)
+
+    for file_formats, params in configs["component_wide"].get(component.id, []):
+        if component.file_format in file_formats:
+            file_format_params.update(params)
+
+    return file_format_params
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("trans", "0044_merge_20250721_1211"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="component",
+            name="file_format_params",
+            field=models.JSONField(
+                blank=True, default=dict, verbose_name="File format parameters"
+            ),
+        ),
+        migrations.RunPython(migrate_addons_config, migrations.RunPython.noop),
```

Comment:
Make the migration elidable when squashing them:

```suggestion
        migrations.RunPython(migrate_addons_config, migrations.RunPython.noop, elidable=True),
```